tests/providers/openai/test_openai.py

```diff
@@ -0,0 +1,39 @@
+from typing import Tuple
+
+import os
+import pytest
+
+from exchange import Text
+from exchange.message import Message
+from exchange.providers.base import Usage
+from exchange.providers.openai import OpenAiProvider
+from .conftest import OPENAI_MODEL, OPENAI_API_KEY
+
+
+@pytest.mark.vcr()
+def test_openai_completion(monkeypatch):
+    # When running VCR tests the first time, it needs OPENAI_API_KEY to call
+    # the real service. Afterward, it is not needed as VCR mocks the service.
+    if "OPENAI_API_KEY" not in os.environ:
+        monkeypatch.setenv("OPENAI_API_KEY", OPENAI_API_KEY)
+
+    reply_message, reply_usage = openai_complete()
+
+    assert reply_message.content == [Text(text="Hello! How can I assist you today?")]
+    assert reply_usage.total_tokens == 27
+
+
+@pytest.mark.integration
+def test_openai_completion_integration():
+    reply = openai_complete()
+
+    assert reply[0].content is not None
```

Comment:
main thing here is that in the integration tests, we can't assert on exact values, where we can (and should) on the unit/vcr tests